src/overlays/actors/ovl_En_Holl/z_en_holl.c

```diff
@@ -24,37 +30,270 @@ const ActorInit En_Holl_InitVars = {
+    Vec3f rotatedPlayerPos;
+    f32 absRotatedPlayerZ;
+
+    if (this->type == EN_HOLL_TYPE_DEFAULT) {
+        u32 actorCtxBitmask =
+            (ACTOR_CONTEXT_UNKC_ODD_10BITS(globalCtx) >> 1) | ACTOR_CONTEXT_UNKC_EVEN_10BITS(globalCtx);
+        u32 zActorBitmask = D_801AED48[EN_HOLL_GET_Z_ACTOR_BITMASK_INDEX(this)];
+        if ((actorCtxBitmask & zActorBitmask) == 0) {
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+        if (this == sThis) {
+            func_800B9010(&this->actor, NA_SE_EV_INVISIBLE_MONKEY - SFX_FLAG);
+        }
+    }
+    if ((globalCtx->sceneLoadFlag != 0) || (globalCtx->unk_18B4A != 0)) {
+        this->opacity = EN_HOLL_OPAQUE;
+    } else {
+        f32 enHollBottom = EN_HOLL_BOTTOM_DEFAULT;
+        f32 enHollWidth = EN_HOLL_HALFWIDTH_DEFAULT;
+
+        EnHoll_SetPlayerSide(globalCtx, this, &rotatedPlayerPos);
+        absRotatedPlayerZ = fabsf(rotatedPlayerPos.z);
+        if (globalCtx->sceneNum == SCENE_IKANA) {
+            enHollBottom = EN_HOLL_BOTTOM_IKANA;
+            enHollWidth = EN_HOLL_HALFWIDTH_IKANA;
+        }
+        if ((enHollBottom < rotatedPlayerPos.y) && (rotatedPlayerPos.y < EN_HOLL_HALFHEIGHT) &&
+            (fabsf(rotatedPlayerPos.x) < enHollWidth) && (absRotatedPlayerZ < sActivationPlaneDistance)) {
+            u32 enHollId = EN_HOLL_GET_ID_AND(this);
+            if (sLoadingPlaneDistance < absRotatedPlayerZ) {
+                if ((globalCtx->roomCtx.prevRoom.num >= 0) && (globalCtx->roomCtx.unk31 == 0)) {
+                    this->actor.room = globalCtx->doorCtx.transitionActorList[enHollId].sides[this->playerSide].room;
+                    if (globalCtx->roomCtx.prevRoom.num == this->actor.room) {
+                        EnHoll_ChangeRooms(globalCtx);
+                    }
+                    func_8012EBF8(globalCtx, &globalCtx->roomCtx);
+                }
+            } else if (EN_HOLL_IS_SCENE_CHANGER(this)) {
+                globalCtx->nextEntranceIndex = globalCtx->setupExitList[EN_HOLL_GET_EXIT_LIST_INDEX(this)];
+                gSaveContext.unk_3DBB = 1;
+                Scene_SetExitFade(globalCtx);
+                globalCtx->sceneLoadFlag = 0x14;
+                globalCtx->unk_1878C(globalCtx);
+            } else {
+                this->actor.room = globalCtx->doorCtx.transitionActorList[enHollId].sides[this->playerSide ^ 1].room;
+                if (globalCtx->roomCtx.prevRoom.num < 0) {
+                    Room_StartRoomTransition(globalCtx, &globalCtx->roomCtx, this->actor.room);
+                    if (this == sThis) {
+                        sThis = NULL;
+                    }
+                } else {
+                    s32 valueToClamp = (absRotatedPlayerZ - sTransparencyPlaneDistance) *
+                                       (EN_HOLL_OPAQUE / (sTranslucencyPlaneDistance - sTransparencyPlaneDistance));
+                    this->opacity = CLAMP(valueToClamp, EN_HOLL_TRANSPARENT, EN_HOLL_OPAQUE);
```

Comment:
Looking throughout this file I still stand by what I was saying in discord-- the defines dont really fit imo

And I think this illustrates the point the best, the alpha is a numeric value that is getting clamped on both edges. The number itself is super important to be able to see here, not the name of what it happens to mean in this context.

And as an aside, I think the variable name can be `planeAlpha` like it is in oot. Its main purpose is to be used in the alpha field of a SetPrimColor after all

Reply:
Both items are addressed in recent pushes, so I'm marking this as resolved. Thanks for the advice!